attachment_queue_email/models/attachment_queue.py

```diff
@@ -0,0 +1,82 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
+import base64
+
+
+class AttachmentQueue(models.Model):
+    _inherit = "attachment.queue"
+
+    fetchmail_attachment_condition_id = fields.Many2one(
+        "fetchmail.attachment.condition",
+        string="FetchMail condition",
+        help="The Fetchmail attachment condition used to create this attachment",
+    )
+    fetchmail_server_id = fields.Many2one(
+        "fetchmail.server",
+        string="Email Server",
+        related="fetchmail_attachment_condition_id.server_id",
+        store=True,
+        readonly=True,
+        help="The email server used to create this attachment",
+    )
+
+    @api.model
+    def _get_attachment_queue_data(self, condition, msg, att):
+        values = {
+            "fetchmail_attachment_condition_id": condition.id,
+            "file_type": condition.file_type,
+            "name": msg.get("subject", att.fname),
+            "sync_date": msg.get("date"),
+            "datas_fname": att.fname,
+            "datas": base64.b64encode(att.content),
+            "state": "pending",
+        }
+        return values
+
+    @api.model
+    def prepare_data_from_basic_condition(self, cond, msg):
+        vals_list = []
+        if str(cond.email_from) in msg.get("from", "") or str(
+            cond.email_subject
```

Comment:
Why is it a `or` condition instead of an `and` ?
If we defined a subject and an email_from, then both condition should be ok to make the import.

Reply:
I thought it could be easier to be able to filter on the "subject" without needing to fill the "from" field (and vice versa).
And in fact this is the case with the native template emails I use in the tests, they have a "from" but no "subject"...

Reply:
@clementmbr 
If the fields are not mandatory, we can consider only one of both conditions is enough, but if both are filled, both condition should be verified.

So yes something like `if (not email_from or email_from in msg) and (not subject or subject in msg...)` should be fine.